### PR TITLE
Move tests to Ginkgo v2

### PR DESCRIPTION
**Pull request by lyarwood**
**What this PR does / why we need it**:

Move tests to Ginkgo v2

**Which issue(s) this PR fixes**: 

Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Functional and unit tests have now moved to Ginkgo v2
```

### CODE REVIEW

File: internal/operands/template-validator/reconcile_test.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 	"github.com/onsi/ginkgo/extensions/table"
```
Comment (akrejcir) [inline internal/operands/template-validator/reconcile_test.go:7]: Can you also move tables to Ginkgo `v2`?

File: go.mod [hunk 2]
```diff
@@ -33,11 +33,13 @@ require (
 	kubevirt.io/client-go v0.58.0
 	kubevirt.io/containerized-data-importer-api v1.53.1
 	kubevirt.io/controller-lifecycle-operator-sdk/api v0.2.4
-	kubevirt.io/qe-tools v0.1.7
+	kubevirt.io/qe-tools v0.1.8
 	kubevirt.io/ssp-operator/api v0.0.0
 	sigs.k8s.io/controller-runtime v0.13.1
 )
 
+require github.com/onsi/ginkgo/v2 v2.1.4
```
Comment (akrejcir) [inline go.mod:41]: When tables are moved to `v2`, you can remove dependency on `github.com/onsi/ginkgo` and move this line to the above `require` block.

File: internal/template-validator/validation/eval.go [hunk 3]
```diff
@@ -39,7 +39,7 @@ var (
 	ErrUnsatisfiedRule      = errors.New("rule is not satisfied")
 )
 
-type Report struct {
+type ValidationReport struct {
```
Comment (akrejcir) [inline internal/template-validator/validation/eval.go:42]: This change is not related to tests.

Comment (lyarwood) [reply internal/template-validator/validation/eval.go:42]: `Report` is already declared by `Ginkgo v2`. Apologies I did mean to leave a note about this in the commit, I'll add one now:

```
$ git checkout origin/master internal/template-validator/validation/eval.go
Updated 1 path from f923d2dd
[08:55:03]lyarwood@hobbiton.yarwood.me.uk:~/redhat/devel/src/k8s/kubevirt/ssp-operator ginkgo-v2# make unittest 
test -s /home/lyarwood/redhat/devel/src/k8s/kubevirt/ssp-operator/bin/controller-gen || GOBIN=/home/lyarwood/redhat/devel/src/k8s/kubevirt/ssp-operator/bin go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.10.0
/home/lyarwood/redhat/devel/src/k8s/kubevirt/ssp-operator/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" "paths={./controllers/..., ./internal/..., ./hack/..., ./webhooks/...}"
cd api && /home/lyarwood/redhat/devel/src/k8s/kubevirt/ssp-operator/bin/controller-gen object:headerFile="../hack/boilerplate.go.txt" "paths=./..."
go fmt ./...
go vet ./...
# kubevirt.io/ssp-operator/internal/template-validator/validation
vet: internal/template-validator/validation/eval.go:42:6: Report already declared through dot-import of package ginkgo ("github.com/onsi/ginkgo/v2")
make: *** [Makefile:155: vet] Error 2
```
